cmd/config.go

```diff
@@ -417,3 +419,129 @@ func overWriteConfig(cmd *cobra.Command, cfg *Config) error {
 	}
 	return err
 }
+
+// ValidatePath checks that a path is valid
+func (c *Config) ValidatePath(p *relayer.Path) (err error) {
+	if err = c.ValidatePathEnd(p.Src); err != nil {
+		return err
+	}
+	if p.Src.Version == "" {
+		return fmt.Errorf("source must specify a version")
+	}
+	if err = c.ValidatePathEnd(p.Dst); err != nil {
+		return err
+	}
+	if _, err = p.GetStrategy(); err != nil {
+		return err
+	}
+	if p.Src.Order != p.Dst.Order {
+		return fmt.Errorf("both sides must have same order ('ORDERED' or 'UNORDERED'), got src(%s) and dst(%s)",
+			p.Src.Order, p.Dst.Order)
+	}
+	return nil
+}
+
+// ValidatePathEnd allow validates provided pathend and returns error for invalid identifiers
+func (c *Config) ValidatePathEnd(pe *relayer.PathEnd) error {
+	if err := pe.ValidateBasic(); err != nil {
+		return err
+	}
+	if pe.ClientID != "" {
+		if err := c.ValidateClient(pe); err != nil {
+			return err
+		}
+	}
+	if pe.ConnectionID != "" {
+		if err := c.ValidateConnection(pe); err != nil {
+			return err
+		}
+	}
+	if pe.ChannelID != "" {
+		if err := c.ValidateChannel(pe); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ValidateClient validates client id in provided pathend
+func (c *Config) ValidateClient(pe *relayer.PathEnd) error {
+	if err := pe.Vclient(); err != nil {
+		return err
+	}
+
+	chain, err := c.Chains.Get(pe.ChainID)
+	if err != nil {
+		return err
+	}
+
+	height, err := chain.QueryLatestHeight()
+	if err != nil {
+		return err
+	}
```

Comment:
This is duplicted in multiple places. May be better to change `ValidateClient` as `ValidateClient(chain, height, pathEnd)` . Also for `ValidateConnection` and `ValidateChannel`